pennylane/interfaces/execution.py

```diff
@@ -357,23 +357,18 @@ def cost_fn(params, x):
         # We must do this now, prior to the interface, to ensure that
         # decompositions with parameter processing is tracked by the
         # autodiff frameworks.
-        for i, tape in enumerate(tapes):
-            tapes[i] = expand_fn(tape)
-

```

Comment:
What if it isn't `adjoint_jacobian`? Then we still need to expand the tapes somewhere.

Reply:
Not sure I understand. This logic is now done on line 371.

Reply:
This logic is moved inside the `if adjoint_jacobian` block. We could have `if gradient_fn="device" without being adjoint jacobian.  So we need to perform this expansion outside the `if adjoint_jacobian` block.

Reply:
Oh I see it wa just moved below.